js/battle.js

```diff
@@ -37,19 +38,22 @@ class LayeredTask extends Task {
     }
 
     onDone() {
-        this.done = true;
+        this.taskProgressBar.style.backgroundColor = lastLayerData.color;
+        this.taskProgressBarFill.style.width = '0%';
+        this.taskProgressBar.getElementsByClassName('name')[0].textContent = 'Defeated';
+        this._done = true;
     }
-}
 
-class Battle extends LayeredTask {
-    constructor(baseData, maxLayers, progressBarId) {
-        super(baseData, maxLayers, progressBarId);
+    isDone(){
+        return this._done;
     }
+}
 
+class Battle extends LayeredTask {
     onDone() {
         super.onDone();
         Events.GameOver.trigger({
-            bossDefeated: this.done,
+            bossDefeated: this._done,
```

Comment:
` this._done` ist jetzt "private" (auch wenn die Sprache das nicht enforced). Per JSDoc 
```
/* @private */
```
kannst du das auch von IDEA erzwingen lassen.

Und dann:
` bossDefeated: this.isDone(),`

Reply:
Wie sieht's mit #done aus? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
Hm... Denke am besten wäre hier "done" komplett zu entfernen und in isDone durch die done-condition zu ersetzen.

Reply:
Wusste nicht, dass sich der Hashtag als private flag etabliert hat. Können wir gerne nutzen 👍